Sources/BraintreePayPal/BTPayPalClient.swift

```diff
@@ -72,6 +72,30 @@ import BraintreeDataCollector
         tokenize(request: request, completion: completion)
     }
 
+    /// Tokenize a PayPal request to be used with the PayPal Vault flow.
+    ///
+    /// - Note: You can use this as the final step in your order/checkout flow. If you want, you may create a transaction from your
+    /// server when this method completes without any additional user interaction.
+    ///
+    /// On success, you will receive an instance of `BTPayPalAccountNonce`; on failure or user cancelation you will receive an error.
+    /// If the user cancels out of the flow, the error code will be `.canceled`.
+    ///
+    /// - Parameter request: A `BTPayPalVaultRequest`
+    /// - Returns: A `BTPayPalAccountNonce` if successful
+    /// - Throws: An `Error` describing the failure
+    @objc(tokenizeWithVaultRequest:completionHandler:)
+    public func tokenize(_ request: BTPayPalVaultRequest) async throws -> BTPayPalAccountNonce {
```

Comment:
Testing these methods made me realize there are some holes in our testing strategy for PayPal and we currently do some weird things to test tokenize. I created ticket DTBTSDK-2512 to revisit our PayPal testing strategy as a whole and maybe use something like the MockASWebSession that we use in SEPA tests to test this more easily. Seems like something we can 🏈  for now though.

For the un-tested methods I tested them in the demo app to ensure they are working as expected.